fish-pepper/run-java-sh/fp-files/run-java.sh

```diff
@@ -70,6 +70,31 @@ get_jar_file() {
   fi
 }
 
+ceiling() {
+  awk -vnumber="$1" -vdiv="$2" '
+    function ceiling(x){
+      return x%1 ? int(x)+1 : x
+    }
+    BEGIN{
+      print ceiling(number/div)
+    }
+  '
+}
+
+# Based on the cgroup limits, figure out the max number of core we should utilize
+max_cores() {
+  local cpu_period_file="/sys/fs/cgroup/cpu/cpu.cfs_period_us"
+  local cpu_quota_file="/sys/fs/cgroup/cpu/cpu.cfs_quota_us"
+  if [ -r "${cpu_period_file}" ]; then
+    local cpu_period="$(cat ${cpu_period_file})"
+
+    if [ -r "${cpu_quota_file}" ]; then
+      local cpu_quota="$(cat ${cpu_quota_file})"
+      ceiling $cpu_quota $cpu_period
```

Comment:
tbh, no really sure about the algo (but I didn't dig into it, too ;-). Could you elaborate a bit (or have a reference) how to come from the CPU sharing quotas to the number of cores that should be used ?